core/src/main/java/org/apache/accumulo/core/fate/Fate.java

```diff
@@ -572,4 +572,10 @@ private void waitForDeadResCleanerShutdown(long start, long timeout, TimeUnit ti
       }
     }
   }
+
+  @Override
+  @SuppressWarnings("deprecation")
+  public final void finalize() {
+    // Prevent finalizer attacks (SpotBugs CT_CONSTRUCTOR_THROW)
+  }
```

Comment:
I'd really rather not add these methods in so many places. If this isn't possible to address another way, I think we can just suppress it. Adding this deprecated method is probably the least good workaround.

Reply:
Yea I agree. Adding the finalize method definitely seems like the worst fix. 

I think the best way to handle things is 
1. if its easy and safe to remove the exception thrown from the constructor, do that (usually not possible)
2. try to make the class final if its obvious that it can be made final
3. add a `@SuppressFBWarnings` tag to the place in the code that the warning is coming from

Do you think that sounds good or would do you think its best to suppress this at module/project level?

Reply:
Avoding finalize here will be helpful in the future https://openjdk.org/jeps/421

Reply:
> Do you think that sounds good or would do you think its best to suppress this at module/project level?

I think trying those 3 fixes, in that order, are probably best. There may be good reasons to suppress at the module/project level, but it should be a very good reason. We have relatively few module/project-wide exceptions, and they are easily articulable, like the comments in the existing `spotbugs/exclude-filter.xml` files.